website/server/models/user/methods.js

```diff
@@ -91,3 +98,17 @@ schema.methods.addComputedStatsToJSONObj = function addComputedStatsToUserJSONOb
 
   return statsObject;
 };
+
+schema.methods.cancelSubscription = async function cancelSubscription () {
+  let plan = this.purchased.plan;
+
+  if (plan.paymentMethod === amazonPayments.constants.PAYMENT_METHOD_AMAZON) {
+    return await amazonPayments.cancelSubscription({user: this});
+  } else if (plan.paymentMethod === stripePayments.constants.PAYMENT_METHOD) {
+    return await stripePayments.cancelSubscription({user: this});
+  } else if (plan.paymentMethod === paypalPayments.constants.PAYMENT_METHOD) {
+    return await paypalPayments.subscribeCancel({user: this});
+  }
+
+  return await payments.cancelSubscription({user: this});
```

Comment:
Yea, it is confusing. The problem is there is a recursive relationship between payments and the other payment providers. Which means we need a third entity. Here the third entity is the User.

You do not want to cal payments.cancelSubscription, because that will not call the api functions with the respective payment provider api. So, to answer you paypal question, if you call the payments.cancelSubscription, you will not correctly cancel the billing date because paypal.getBillingDate will not be called.

There is a better way to organize this, but that should wait until the next refactor. I can put it on my list for right after this PR, but I don't want to hold up the PR because it is already large.